basicapp/managedapps.go

```diff
@@ -0,0 +1,216 @@
+package basicapp
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/giantswarm/helmclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/e2etests/basicapp/legacyresource"
+)
+
+type Config struct {
+	Clients    Clients
+	HelmClient *helmclient.Client
+	Logger     micrologger.Logger
+
+	App            Chart
+	ChartResources ChartResources
+}
+
+type ManagedApps struct {
+	clients    Clients
+	helmClient *helmclient.Client
+	logger     micrologger.Logger
+	resource   *legacyresource.Resource
+
+	chart          Chart
+	chartResources ChartResources
+}
+
+func New(config Config) (*ManagedApps, error) {
+	var err error
+
+	if config.HelmClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.HelmClient must not be empty", config)
+	}
+	if config.Clients == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Clients must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	err = config.App.Validate()
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	err = config.ChartResources.Validate()
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var resource *legacyresource.Resource
+	{
+		c := legacyresource.Config{
+			HelmClient: config.HelmClient,
+			Logger:     config.Logger,
+			Namespace:  config.App.Namespace,
+		}
+
+		resource, err = legacyresource.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	ms := &ManagedApps{
+		clients:    config.Clients,
+		helmClient: config.HelmClient,
+		logger:     config.Logger,
+		resource:   resource,
+
+		chart:          config.App,
+		chartResources: config.ChartResources,
+	}
+
+	return ms, nil
+}
+
+func (ms *ManagedApps) Test(ctx context.Context) error {
```

Comment:
```suggestion
func (b *BasicApp) Test(ctx context.Context) error {
```

This is my bad but I think just `b` is enough.